File: apps/src/chat/backend/packets/websocket/update_status_packet.py
from ..packet import Packet

class UpdateStatusPacket(Packet):

    def __init__(self, username: str):
        self._username = username

    def to_data(self):
        return {
            "username": self._username,
            "type": "update_status"
        }

    @staticmethod
    def from_data(data):
        return UpdateStatusPacket(
            data["username"]
        )

File: apps/src/chat/frontend/js/packets/websocket/UpdateStatusPacket.js
export default class UpdateStatusPacket {

    constructor(username = ""){
        this.username = username;
    }

    toData(){
        return {
            username: this.username,
            type: "global_join"
        }
    }

    static fromData(data){
        return new UpdateStatusPacket(data["username"]);
    }

    getUsername(){
        return this.username;
    }
}